proposals/NNNN-atf-script-runner.md

```diff
@@ -0,0 +1,41 @@
+# ATF Script Runner
+
+* Proposal: [SDL-NNNN](NNNN-atf-script-runner.md)
+* Author: [Dmytro Boltovskyi](https://github.com/dboltovskyi)
+* Status: **Awaiting review**
+* Impacted Platforms: [ATF]
+
+## Introduction
+
+ATF is a great tool for both developers and testers that help test any change in SDL.
+But it lacks a powerful test script runner.
+
+## Motivation
+
+Currently ATF is unable to:
+
+  - run multiple test scripts
+  - run test script in parallel
+  - create test reports in standard format
+
+## Proposed solution
+
+The purpose of this proposal is to develop extended script runner as part of ATF functionality with the following features:
+
+  - An option to run single test script
+  - An option to run batch of test scripts
+  - An option to run test set
+  - Possibility to run test scripts in parallel (in separate threads)
```

Comment:
@dboltovskyi 
My previous comment on the same point is still actual even if you'll run tasks in separate threads. How are you going to use SDL in that case so we can avoid concurrent usage of devices or SDL internal state changes due to multiple tests run?

Reply:
@dev-gh Maybe my comment was missed. Let me duplicate it here:
> Yep - with the same instance of SDL it's impossible. But the idea is to use SDL Watchdog (new feature in separate proposal) that will provide a new SDL instance for test script from the pull of available instances.

So each test script from test set will be run within separate SDL instance of a certain type (e.g.: RC=ON, PROPRIETARY)